python/_restclient/hydrotools/_restclient/utilities.py

```diff
@@ -0,0 +1,181 @@
+            if isinstance(self.valid_value, IterableNonStringLike)
+            else frozenset([self.valid_value])
+        )
+
+        if isinstance(self.key, Alias):
+            # Alias is passed as key, use contents to extend into new instance
+            self.__dict__["valid_value"] = self.valid_value | self.key.valid_value
+            self.__dict__["key"] = self.key.key
+
+    def get(self, value: Hashable) -> Union[Any, None]:
+        """Get key given a valid alias value. If a valid key is not provided, return
+        None.
+
+        Parameters
+        ----------
+        value : Hashable
+            Valid alias value
+
+        Returns
+        -------
+        Union[Any, None]
+           alias key if valid value, else None
+        """
+        if value in self:
+            return self.key
+
+        return None
+
+    def __contains__(self, value) -> bool:
+        return value in self.valid_value
+
+    def __getitem__(self, value) -> Any:
+        key = self.get(value)
+
+        if key is None:
+            raise ValueError(
+                "Invalid value %s. Valid values are %s" % (value, self.valid_value)
+            )
+
+        return key
+
+    def __str__(self):
+        return str(self.key)
+
+    def __repr__(self):
+        return f"{str(self.key)}: {str(self.valid_value)}"
+
+
+class AliasGroup:
+    """Create a group of Alias objects that are treated like a single Alias. This
+    comes in handy to cut down on conditional statements when developing for example,
+    rest api libraries. This is clearest shown through an example.
+
+    Example:
+        base_url = "www.api.org"
+
+        # In the below, both path-1 and path-2 are different entities.
+        # api.org/path-1/cool-feature
+        # api.org/path-2/cool-feature
+
+        path_1 = Alias("path-1", [1, "1"])
+        path_2 = Alias("path-2", [2, "2"])
+        path_group = GroupAlias([path_1, path_2])
+
+        def get_cool_feature(path: Union[int, str]):
+            path = path_group[path] # ValueError thrown if invalid path
+
+            url = f"{base_url}/{path}/cool-feature"
+
+            response = requests.get(url)
+            return response.json()
+    """
+
+    def __init__(self, alias: List[Alias]) -> None:
+
+        self._option_groups = (
+            frozenset(alias)
+            if isinstance(alias, IterableNonStringLike)
+            else frozenset([alias])
+        )
+
+        symmetric_differences, union = frozenset(), frozenset()
+        self.option_map = {}
+
+        for member in self._option_groups:
+            symmetric_differences = member.valid_value ^ symmetric_differences
+            union = member.valid_value | union
+
+            for valid_value in member.valid_value:
+                self.option_map[valid_value] = member
+
+        duplicate_value = union - symmetric_differences
+
+        if duplicate_value:
+            raise ValueError(f"Repeated valid_value {duplicate_value} not allowed")
```

Comment:
This is not intuitive, since this is on the alias values of the passed in alias features.  The messaging and doc string should make this distinction better.

Reply:
Yeah having read this again I agree haha. Ill update the message to make it more clear.